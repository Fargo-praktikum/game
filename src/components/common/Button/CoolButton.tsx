import React, {MouseEventHandler} from "react";
import "./CoolButton.scss";

type CoolButtonType = {
    clickHandler: MouseEventHandler,
    text: string,
    isDisabled?: boolean,
}

const CoolButton = ({clickHandler, text, isDisabled}: CoolButtonType) => {

  return (
    <button className="cool-button"
            disabled={isDisabled}
            onClick={clickHandler}
    >
      <span>{text}</span>
    </button>
  );
};

export default CoolButton;
