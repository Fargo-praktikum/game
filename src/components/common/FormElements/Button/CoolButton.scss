@import "../../../Forum/forumVar";

.cool-button {
    padding: 5px 30px;
    border: 0;
    border-radius: 8px;
    color: $white;
    font-weight: 500;
    font-size: 18px;
    font-style: normal;
    line-height: 24px;
    background: $blue;
    outline: none;
    cursor: pointer;
    transition: all 0.2s linear;
    user-select: none;

    &:hover {
        background: $lightblue;
    }

    &:disabled {
        color: $dark-gray;
        background: $gray-locked;
        cursor: default;
    }

    &:active {
        background: $darkblue;
        transition: all 0s linear;
    }
}
