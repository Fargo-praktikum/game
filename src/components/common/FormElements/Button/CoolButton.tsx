import React, {MouseEventHandler} from "react";
import "./CoolButton.scss";

type CoolButtonType = {
    text: string;
    isDisabled?: boolean;
    type?: "submit" | "reset" | "button";
    clickHandler?: MouseEventHandler;
}

const CoolButton = ({clickHandler, text, isDisabled, type}: CoolButtonType) => {

  return (
    <button className="cool-button"
            disabled={isDisabled}
            onClick={clickHandler}
            type={type}
    >
      <span>{text}</span>
    </button>
  );
};

export default CoolButton;
