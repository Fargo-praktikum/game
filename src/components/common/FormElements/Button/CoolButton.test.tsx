import React from "react";

import renderer from "react-test-renderer";
import CoolButton from "./CoolButton";

describe("Проверка CoolButton", () => {
    it("корректная отрисовка", () => {
        const tree = renderer
            .create(<CoolButton
                text="Нажми меня"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("наличие всех свойств", () => {
        const tree = renderer.create(<CoolButton
            text="Нажми меня"
            clickHandler={() => {
                console.log("нажал кнопку");
            }}
            isDisabled={false}
            type={"reset"}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



