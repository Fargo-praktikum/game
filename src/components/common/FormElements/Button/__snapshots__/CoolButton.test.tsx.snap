// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка CoolButton корректная отрисовка 1`] = `
<button
  className="cool-button"
>
  <span>
    Нажми меня
  </span>
</button>
`;

exports[`Проверка CoolButton наличие всех свойств 1`] = `
<button
  className="cool-button"
  disabled={false}
  onClick={[Function]}
  type="reset"
>
  <span>
    Нажми меня
  </span>
</button>
`;
